runtime/lua/vim/keymap.lua

```diff
@@ -59,7 +61,12 @@ function keymap.set(mode, lhs, rhs, opts)
   if is_rhs_luaref and opts.expr and opts.replace_keycodes ~= false then
     local user_rhs = rhs
     rhs = function ()
-      return vim.api.nvim_replace_termcodes(user_rhs(), true, true, true)
+      local res = user_rhs()
+      if not res then
+        return '<nop>'
```

Comment:
Shouldn't this be an empty string instead?

Reply:
When I use an empty string I experience different behaviour to `<nop>`.

Reply:
This is currently what I'm playing with:

```lua
  map('n', ']c', function()
    if vim.wo.diff then return ']c' end
    vim.schedule(gitsigns.next_hunk)
    return '<nop>'
  end, {expr=true})

  map('n', '[c', function()
    if vim.wo.diff then return '[c' end
    vim.schedule(gitsigns.next_hunk)
    return '<nop>'
  end)
```

Reply:
Using `''` doesn't redraw the screen after callbacks have run, for some reason using `<nop>` works.

Reply:
What???

Reply:
Note that `vim.schedule` is only needed because the nav functions run `vim.cmd [[ normal! m' ]]` which isn't allowed in expr mappings.

Reply:
The use `vim.api.nvim_replace_termcodes('<Ignore>', true, true, true)`

Reply:
Ah, good to know. So why does `<Nop>` work?

Reply:
Ok, fair. I'll update to use the empty string.

Reply:
It can be hard to decide between empty string and `<Ignore>`, as it is hard to know what people actually use expr mappings for.

Reply:
Well, the current behaviour is:

```lua
E5108: Error executing lua: Expected lua string
stack traceback:
        [C]: at 0x56025e6d9a86
```

Which I'm not sure we want.